File: 2026/aed-1/cormen/2.1/insertion_sort.rs
fn insertion_sort(a: &mut [i32], n: usize) {
    for i in 1..n {
        let key = a[i];
        let mut j = i;
        while j > 0 && a[j - 1] > key {
            a[j] = a[j - 1];
            j -= 1;
        }
        a[j] = key;
    }
}

fn main() {
    let mut a = [4, 3, 1, 6, 10];
    let n = a.len();

    println!("Unsorted.");

    for i in &a {
        println!("{}", i);
    }

    insertion_sort(&mut a, n);

    println!("Sorted.");

    for i in &a {
        println!("{}", i);
    }
}

File: 2026/aed-1/cormen/2.1/insertion_sort.cpp
#include <iostream>

void insertion_sort(int *A, int n) {
  for (int i = 1; i < n; i++) {
    int key = A[i];
    int j = i - 1;

    while (j >= 0 && A[j] > key) {
      A[j + 1] = A[j];
      j = j - 1;
    }

    A[j + 1] = key;
  }
}

int main() {
  int A[5] = {4, 3, 1, 6, 10};
  int n = sizeof(A) / sizeof(int);

  std::cout << "Unsorted." << '\n';

  for (int i = 0; i < n; i++) {
    std::cout << A[i] << '\n';
  }

  insertion_sort(A, n);

  std::cout << "Sorted." << '\n';

  for (int i = 0; i < n; i++) {
    std::cout << A[i] << '\n';
  }

  return 0;
}
